android/java/org/chromium/chrome/browser/rewards/RewardsPageActivity.java

```diff
@@ -0,0 +1,168 @@
+
+    @Override
+    public boolean supportsAppMenu() {
+        return false;
+    }
+
+    @Override
+    public void performPostInflationStartup() {
+        super.performPostInflationStartup();
+
+        View toolbarContainer = findViewById(R.id.toolbar_container);
+        if (toolbarContainer != null) {
+            toolbarContainer.setVisibility(View.GONE);
+        }
+
+        FrameLayout.LayoutParams layoutParams =
+                new FrameLayout.LayoutParams(
+                        FrameLayout.LayoutParams.WRAP_CONTENT,
+                        FrameLayout.LayoutParams.WRAP_CONTENT);
+        layoutParams.gravity = Gravity.TOP | Gravity.START;
+        layoutParams.setMargins(CLOSE_BUTTON_MARGIN, CLOSE_BUTTON_MARGIN, 0, 0);
+
+        ViewGroup parentView = findViewById(android.R.id.content);
+        ImageView closeImg = new ImageView(RewardsPageActivity.this);
+        closeImg.setPadding(
+                CLOSE_BUTTON_PADDING,
+                CLOSE_BUTTON_PADDING,
+                CLOSE_BUTTON_PADDING,
+                CLOSE_BUTTON_PADDING);
+
+        Drawable drawable =
+                ResourcesCompat.getDrawable(getResources(), R.drawable.rewards_page_close_bg, null);
+        closeImg.setBackground(drawable);
+        closeImg.setImageResource(R.drawable.ic_baseline_close_24);
+        closeImg.setColorFilter(
+                ContextCompat.getColor(RewardsPageActivity.this, android.R.color.black),
+                android.graphics.PorterDuff.Mode.SRC_IN);
+        closeImg.setOnClickListener(
+                button -> {
+                    finish();
+                });
+        parentView.addView(closeImg, layoutParams);
+    }
+
+    public static void showPage(Context context, String url) {
+        Intent intent = new Intent();
+        intent.setAction(Intent.ACTION_VIEW);
+        intent.setClassName(context, RewardsPageActivity.class.getName());
+        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        intent.putExtra(CustomTabsIntent.EXTRA_TITLE_VISIBILITY_STATE, CustomTabsIntent.NO_TITLE);
+        intent.putExtra(CustomTabsIntent.EXTRA_ENABLE_URLBAR_HIDING, false);
+        intent.putExtra(
+                CustomTabsIntent.EXTRA_COLOR_SCHEME,
+                ColorUtils.inNightMode(context) ? COLOR_SCHEME_DARK : COLOR_SCHEME_LIGHT);
+        intent.setData(Uri.parse(url));
+        intent.setPackage(context.getPackageName());
+        intent.putExtra(
+                EXTRA_UI_TYPE, BrowserServicesIntentDataProvider.CustomTabsUiType.INFO_PAGE);
+        intent.putExtra(Browser.EXTRA_APPLICATION_ID, context.getPackageName());
+        if (!(context instanceof Activity)) intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        IntentUtils.addTrustedIntentExtras(intent);
+
+        context.startActivity(intent);
+    }
+
+    @CalledByNative
+    private static void openURLForRewardsPageActivity(String url) {
+        TabUtils.openURLWithBraveActivity(url);
```

Comment:
this method is called `openURLForRewardsPageActivity`, but appears to allow any url

Reply:
if we do actually need this and there is no existing upstream functionality I don't think it makes sense to create a rewards specific wrapper for this. We should just expose `TabUtils.openURLWithBraveActivity` directly to cpp